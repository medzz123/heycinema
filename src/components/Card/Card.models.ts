export interface CardProps {
  image: string;
  name: string;
  rating: string;
  releaseYear: string;
}
