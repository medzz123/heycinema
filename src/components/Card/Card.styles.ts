import styled from 'styled-components';

export const CardContainer = styled.div`
  border-radius: 10px;
  background-color: white;
  display: flex;
  width: 400px;
  height: 125px;
`;

export const ImageContainer = styled.div`
  width: 35%;
`;

export const ContentContainer = styled.div`
  padding: 10px;
  width: 65%;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
`;

export const MovieName = styled.h1`
  margin: 0 0 2px 0;
  color: ${(p) => p.theme.primary};
`;

export const MovieRating = styled.p`
  margin: 0;
  color: ${(p) => p.theme.text.info};
`;

export const MovieReleaseDate = styled.p`
  margin: 0;
  color: ${(p) => p.theme.tertiary};
`;
