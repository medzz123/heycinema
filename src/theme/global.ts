import { createGlobalStyle, css } from 'styled-components';

// Inside css for auto formatting with eslint
const styles = css`
  html {
    font-family: 'Mulish', sans-serif;
    scroll-behavior: smooth;
  }

  body,
  html {
    margin: 0;
    padding: 0;
    background-color: grey;
  }

  * {
    box-sizing: border-box;
  }

  ul {
    list-style: none;
  }

  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    font-size: 18px;
  }

  a {
    text-decoration: none;
    color: inherit;
  }

  p {
    font-size: 16px;
  }
`;

const GlobalStyles = createGlobalStyle`
  ${styles}
`;

export default GlobalStyles;
