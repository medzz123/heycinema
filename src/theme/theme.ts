export const defaultTheme = {
  primary: '#7FAAE4',
  secondary: '#834DB7',
  tertiary: '#34283F',
  text: {
    info: 'grey',
  },
};
