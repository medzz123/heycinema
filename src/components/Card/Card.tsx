import React from 'react';

import { CardProps } from './Card.models';
import {
  CardContainer,
  ContentContainer,
  ImageContainer,
  MovieName,
  MovieRating,
  MovieReleaseDate,
} from './Card.styles';

const Card: React.FC<CardProps> = (props) => {
  const { name, image, rating, releaseYear } = props;
  return (
    <CardContainer>
      <ImageContainer url={image} />
      <ContentContainer>
        <div>
          <MovieName>{name}</MovieName>
          <MovieRating>{rating}</MovieRating>
        </div>
        <MovieReleaseDate>{releaseYear}</MovieReleaseDate>
      </ContentContainer>
    </CardContainer>
  );
};

export default Card;
