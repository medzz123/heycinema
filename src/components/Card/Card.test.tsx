import { defaultTheme } from '@theme/theme';
import React from 'react';
import TestRenderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';

import Card from './Card';

describe('<Card />', () => {
  it('should render without throwing an error and match snapshot', () => {
    const testRenderer = TestRenderer.create(
      <ThemeProvider theme={defaultTheme}>
        <Card
          name="Test Sample"
          image="www.google.com"
          rating="0.3"
          releaseYear="2015"
        />
      </ThemeProvider>
    );
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
