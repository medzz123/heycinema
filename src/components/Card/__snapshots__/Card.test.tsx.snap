// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should render without throwing an error and match snapshot 1`] = `
<div
  className="sc-bdnylx cNEHbd"
>
  <div
    className="sc-gtssRu hlwygE"
  />
  <div
    className="sc-dlnjPT cKmuUl"
  >
    <div>
      <h1
        className="sc-hKFyIo iVOQiM"
      >
        Test Sample
      </h1>
      <p
        className="sc-eCApGN UjWI"
      >
        0.3
      </p>
    </div>
    <p
      className="sc-jSFkmK cJhice"
    >
      2015
    </p>
  </div>
</div>
`;
